ansible/configs/sap-smart/post_software.yml

```diff
@@ -85,24 +85,44 @@
         name: satellite-tools-6.7-for-rhel-8-x86_64-e4s-rpms
         state: enabled
 
-    - name: upgrade all packages
-      yum:
-        name: '*'
-        state: latest
-
     - name: Reboot instances
       reboot:
 
-    # - name: Install katello-agent and compliance packages
-    #   yum:
-    #     name: "{{ item }}"
-    #     state: present
-    #   loop:
-    #     - katello-agent
+    - name: Install katello-agent and compliance packages
+      yum:
+        name: "{{ item }}"
+        state: present
+      loop:
+        - katello-agent
```

Comment:
You don't need a loop the module accepts a list.
You can also favor [`package`](https://docs.ansible.com/ansible/latest/modules/package_module.html) vs `yum`

```yaml
    - name: Install katello-agent and compliance packages
      package:
        name:
          - katello-agent
        state: present
```

Reply:
Done, keeping `yum` for consistency across the whole configuration